test/Makefile

```diff
@@ -12,16 +12,19 @@ fake-next-minor-rpm:
 	RPMBUILD_DIR=$(RPMBUILD_DIR) \
 		$(MAKE) -C $(REPO) rpm
 
-.PHONY: local-main-rpm
-local-main-rpm: TMP_REPO !=mktemp -d
-local-main-rpm: RPMBUILD_DIR :=$(REPO)/_output/rpmbuild-main/
-local-main-rpm: COMMIT !=git rev-parse main
-local-main-rpm:
-	git clone --single-branch --branch main --no-hardlinks "$(REPO)" "$(TMP_REPO)"
-	cd "$(TMP_REPO)" && \
-		MICROSHIFT_VERSION="$(MAJOR).$(MINOR).$(PATCH)-main-$(COMMIT)" \
-		RPMBUILD_DIR="$(RPMBUILD_DIR)"\
-		$(MAKE) rpm
+# build-base-branch
+# By default, BASE_BRANCH is derived from the microshift/Makefile.version.$(uname -i).var string.
+# The BASE_BRANCH is pulled from the remote repo.  This is because OpenShift-CI clones the user's
+# source from their fork, which may or may not include the upstream release branches.
```

Comment:
Wow. That's a surprise.

Reply:
Out of curiosity I looked at output of [src job](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_microshift/2158/pull-ci-openshift-microshift-main-microshift-metal-tests/1687109070437224448/artifacts/build-logs/src.log) and found out following:

CLONEREFS are params for creating `src` image - most important things are `base_sha` which points to main's commit and `refs.pulls.sha` which actually exists on the openshift/microshift repo (when you navigate to PR's commit it doesn't forward you to a fork).
We can even see that it will merge PR onto a base branch (we can sometimes get an error at this stage when github (tide?) doesn't notice PR should be rebased).

But let's get back to matter at hand: looking at the log, what we get in `src` is in fact missing real `main` (or `release-4.YY`) because it's base+merged PR. And this is very sparse (shallow?) clone (git init + fetch SHA), so probably that's why it has no extra branches.
I wouldn't say it clones the fork, but it's not obvious what's in the repo, but other than that good job figuring this out!

```
ENV "BUILD_LOGLEVEL"="0" "CLONEREFS_OPTIONS"="
{
  "src_root": "/go",
  "log": "/dev/null",
  "git_user_name": "ci-robot",
  "git_user_email": "ci-robot@openshift.io",
  "refs": [
    {
      "org": "openshift",
      "repo": "microshift",
      "repo_link": "https://github.com/openshift/microshift",
      "base_ref": "main",
      "base_sha": "f7cb02905ad495c2a47c2c44cfa6f25ded592812",
      "base_link": "https://github.com/openshift/microshift/commit/f7cb02905ad495c2a47c2c44cfa6f25ded592812",
      "pulls": [
        {
          "number": 2158,
          "author": "copejon",
          "sha": "84acc4b9bef0d1e7a8e75e8f82df9d606f40418b",
          "head_ref": "upgrade-test-use-head",
          "link": "https://github.com/openshift/microshift/pull/2158",
          "commit_link": "https://github.com/openshift/microshift/pull/2158/commits/84acc4b9bef0d1e7a8e75e8f82df9d606f40418b",
          "author_link": "https://github.com/copejon"
        }
      ]
    }
  ],
  "fail": true
}
"

git init
git config user.name ci-robot
git config user.email ci-robot@openshift.io
git fetch https://github.com/openshift/microshift.git --tags --prune
git fetch https://github.com/openshift/microshift.git f7cb02905ad495c2a47c2c44cfa6f25ded592812
git checkout f7cb02905ad495c2a47c2c44cfa6f25ded592812
git branch --force main f7cb02905ad495c2a47c2c44cfa6f25ded592812
git checkout main
git fetch https://github.com/openshift/microshift.git 84acc4b9bef0d1e7a8e75e8f82df9d606f40418b
git merge --no-ff 84acc4b9bef0d1e7a8e75e8f82df9d606f40418b
git submodule update --init --recursive
```

Reply:
Thanks! Totally agree too - it's not a clone, but just the ref, as you point out.  Since it needs a rebase, I'll tweak the comment for accuracy